ACM 2.12.3 Errata (#7737)
open-cluster-management/rhacm-docs: release_notes/acm_errata.adoc

*Important:* Cluster lifecycle components and features are within the {mce-short}, which is a software operator that enhances cluster fleet management. Release notes for {mce-short}-specific features are found in at link:../clusters/release_notes/mce_release_notes.adoc#mce-release-notes[Release notes for Cluster lifecycle with {mce-short}].

== Errata 2.12.3

* Delivers updates to one or more product container images.

> * Fixes the console errors caused by a policy having an invalid structure. (link:https://issues.redhat.com/browse/ACM-16238[ACM-16238])

Author: @yiraeChristineKim

> * Stores the correct {ocp-short} version to the database when a cluster gets imported to {acm-short}. (link:https://issues.redhat.com/browse/ACM-17553[ACM-17553])

Author: @jhernand

Reviewer: Looks good to me.

> * Allows virtual machine actions on remote clusters with signed certificates. (link:https://issues.redhat.com/browse/ACM-17065[ACM-17065])

Author: @KevinFCormier

> * Preserves the node object even when the `ClusterInstance` gets deleted from the cluster management hub. (link:https://issues.redhat.com/browse/ACM-17877[ACM-17877])

Author: @jhernand

Reviewer: Looks good to me.

> * Prevents the same metric from getting reported multiple times and triggering a `PrometheusDuplicateTimestamps` alert by only reporting that metric one time with a count of the number of {ocp} templates with violations. (link:https://issues.redhat.com/browse/ACM-16014[ACM-16014])

Author: @JustinKuli

Reviewer 2: LGTM

> * Prevents the same metric from getting reported multiple times and triggering a `PrometheusDuplicateTimestamps` alert by only reporting that metric one time with a count of the number of {ocp} templates with violations. (link:https://issues.redhat.com/browse/ACM-16014[ACM-16014])

Reviewer 3: ```suggestion
* Prevents the same metric from getting reported multiple times and triggering a `PrometheusDuplicateTimestamps` alert by only reporting that metric one time with a count of the number of {ocp-short} templates with violations. (link:https://issues.redhat.com/browse/ACM-16014[ACM-16014])
```

> * Stops the metrics collector from degrading the addon status when the collector receives the HTTP 409 error code from the hub cluster. (link:https://issues.redhat.com/browse/ACM-16493[ACM-16493])

Author: @thibaultmg

Reviewer 4: Looks good. We could also add that when the hub responds with a 409, only duplicates from the payload are rejected, other metric samples are ingested correctly.

> * Stops the metrics collector from degrading the addon status when the collector receives the HTTP 409 error code from the hub cluster. When the hub cluster responds with a 409 error code, it correctly loads metric samples and rejects duplicates from the payload.  (link:https://issues.redhat.com/browse/ACM-16493[ACM-16493])

Reviewer 3: ```suggestion
* Stops the metrics collector from degrading the addon status when the collector receives the HTTP 409 error code from the hub cluster, which correctly loads metric samples and rejects duplicates from the payload.  (link:https://issues.redhat.com/browse/ACM-16493[ACM-16493])
```

> 

Reviewer 3: His feedback:

We could also add that when the hub responds with a 409, only duplicates from the payload are rejected, other metric samples are ingested correctly.

Stops the metrics collector from degrading the add-on status when the collector receives the HTTP 409 error code from the hub cluster, which correctly loads metric samples and rejects duplicates from the payload.

Reviewer 4: LGTM

> * Corrects the `ACMRemoteWriteError` alert so that it correctly reports when there is a 20% error rate of remote writes. (link:https://issues.redhat.com/browse/ACM-18000[ACM-18000])

Author: @jacobbaungard

Reviewer 3: Please check for how we write percentages?

Reviewer 5: Looks fine to me!

> * Corrects the `ACMRemoteWriteError` alert so that it correctly reports when there is a 20% error rate of remote writes. (link:https://issues.redhat.com/browse/ACM-18000[ACM-18000])

Reviewer 3: ```suggestion
* Corrects the `ACMRemoteWriteError` alert so that it accurately reports when there is a 20% error rate of remote writes. (link:https://issues.redhat.com/browse/ACM-18000[ACM-18000])
```

Reviewer 3: (since we use correct twice)
